Chap_API_Server.tex

```diff
@@ -223,6 +215,37 @@ \subsection{\code{PMIx_server_register_nspace}}
 
 Pass job-related information to the \ac{PMIx} server library for distribution to local client processes.
 
+The \refarg{info} argument is expected to be a set of
+\begin{itemize}
+    \item namespace-level key-values: 
+    \begin{itemize}
+        \item required: \refattr{PMIX_NSPACE}, \refattr{PMIX_JOBID}, \refattr{PMIX_NODE_LIST}, \refattr{PMIX_UNIV_SIZE}, \refattr{PMIX_JOB_SIZE}, \refattr{PMIX_MAX_PROCS}, \refattr{PMIX_NUM_NODES}, \refattr{PMIX_NODE_MAP}, \refattr{PMIX_PROC_MAP}
+        \item optional: \refattr{PMIX_SERVER_NSPACE}, \refattr{PMIX_SERVER_RANK}, \refattr{PMIX_NPROC_OFFSET}, \refattr{PMIX_APPLDR}, \refattr{PMIX_SESSION_ID}, \refattr{PMIX_ALLOCATED_NODELIST}, \refattr{PMIX_JOB_NUM_APPS}, \refattr{PMIX_MAPBY}, \refattr{PMIX_RANKBY}, \refattr{PMIX_BINDTO}
+    \end{itemize}
+    \item node-level key-values: 
+    \begin{itemize}
+        \item required: \refattr{PMIX_LOCAL_SIZE}, \refattr{PMIX_LOCAL_PEERS}. \refattr{PMIX_LOCAL_CPUSETS}
+        \item optional: \refattr{PMIX_AVAIL_PHYS_MEMORY}, \refattr{PMIX_HWLOC_XML_V1}, \refattr{PMIX_HWLOC_XML_V2}, \refattr{PMIX_LOCALLDR}, \refattr{PMIX_NODE_SIZE}, \refattr{PMIX_LOCAL_PROCS}
+    \end{itemize}
+    \item Multiple complex key-values, each named \refattr{PMIX_PROC_DATA}, encapsulating per-process information.
+
+          \pastePRRTEAttributeItem{PMIX_PROC_DATA}
+
+          \refattr{PMIX_PROC_DATA} must have type \emph{PMIX_DATA_ARRAY} (\refstruct{pmix_data_array_t}) with elements of type \emph{PMIX_INFO} (\refstruct{pmix_info_t}).
+          The set of information enclosed in each of such keys is provided below (NOTE that the first key-value must be rank of the process as per \refattr{PMIX_PROC_DATA} description).
+    \begin{itemize}
+        \item application-level key-values (in case of multiple applications):
+        \begin{itemize}
+            \item required: \refattr{PMIX_APPNUM}, \refattr{PMIX_APPLDR}, \refattr{PMIX_APP_SIZE}, \refattr{PMIX_APP_RANK}
+        \end{itemize}
+        \item process-level key-values:
+        \begin{itemize}
+            \item required: \refattr{PMIX_RANK}, \refattr{PMIX_LOCAL_RANK}, \refattr{PMIX_NODE_RANK}, \refattr{PMIX_NODEID}
+            \item optional: \refattr{PMIX_PROCID}, \refattr{PMIX_GLOBAL_RANK}, \refattr{PMIX_HOSTNAME}
+        \end{itemize}
+    \end{itemize}
+\end{itemize}
```

Comment:
I'm not sure I understand how this differs from the prior itemized list of required and optional values. We certainly don't want to list them all twice, so perhaps we just need to look at organization?

Reply:
I'd personally prefer to have a summary of the keys and the groups they belong to.
The original description is less visual from my perspective also given that it also provides meanings of each key.
IMO grouping by key classes (job-,node-,app-,proc-level) and then by required/optional would improve interpretation. Because this is what implementor will need to do - group the keys.

I also think that we need to explicitly say that process information has to be grouped. Otherwise PMIX_PROC_DATA key is meaningless in the standard. It is defined but never used.

I recognize that other people may think in different way so maybe it is a good thing to provide both classifications.

Reply:
I'm afraid we already decided quite a while back to group attributes by "required" vs "optional", and it is too late to revisit that decision. What we can do is group things within those categories, and I'm happy to take a crack at it.

Reply:
> What we can do is group things within those categories, and I'm happy to take a crack at it.

That may be ok, but my original point was to have just a key list without explanations to provide a summary. If we will re-group it won't be a summary.

Reply:
Well, like I said, we already decided on a format - and we really need to stick with it. We have so far gotten very positive reviews on it 😄

Reply:
But I'm keeping the original grouping providing an additional summary. I don't think it confronts the original format.

Reply:
> Well, like I said, we already decided on a format - and we really need to stick with it. We have so far gotten very positive reviews on it 😄

One negative from me

Reply:
Though I agree that this grouping helps a bunch, but for the cases where there is a lot of possible parameters it is a good idea to provide a summary. And that's what I did.

Reply:
Understood - sorry you don't like the format, but it was decided upon by those participating at the time. We can't change it now, but we can try to provide some clarifications in the hope it meets the general needs.

Reply:
You don't quite answer my question.
I'm not arguing about changing the format. Only proposing to provide additional clarification. Why is that not appropriate?